queries/python/indents.scm

```diff
@@ -27,6 +27,8 @@
   (concatenated_string)
 ] @indent
 
+; TODO: this predicate is not executed often enough to prevent accidental matches
+;((expression_statement) @indent (#lua-match? "\\\\\\n" @indent))
```

Comment:
this fixes the indentation after `\`, but seems to match unrelated statements which seems to be a bug in the Nvim's query implementation @MunifTanjim do you see a reason why this shouldn't work? The actual Lua query for `find` is `\\\n` so a verbatim `\` followed by a `\n`. All `\` have to be escaped in the query file to not enter escape sequences in the string in the query file

#2407 #2376  also reported that `match` accidentally match on unrelated queries.

Reply:
I don't think it's about the query. Need to check which initial node is detected by the indent module here: https://github.com/nvim-treesitter/nvim-treesitter/blob/58a4897e6d5f21a9178ae0f802d47af26d67a219/lua/nvim-treesitter/indent.lua#L103-L107

Reply:
But adding this query that does not match in the file at all shouldn't change highlighting behavior, right?

Reply:
Oh I see... this query is incorrect I think, `((expression_statement) @indent (#lua-match? "\\\\\\n" @indent))` is causing runtime error (it does so in playground). It should be: `((expression_statement) @indent (#lua-match? @indent "\\\\\\n"))`

Reply:
Oops, thanks for noting this

Reply:
btw, for this

```python
a = 2 \
    + 2
```

when you enter a newline after L1


```python
a = 2 \

    + 2
```

And parse the tree, the `expression_statement` in L1 just becomes `a = 2` (the `\\n` is not part of it anymore). So that query will be useless.

Reply:
But the capture won't fix the problem anyway...